spec.emu

```diff
@@ -35,19 +35,22 @@ markEffects: true
       <p>When removing the common leading indentation, lines which contain only whitespace will have all whitespace removed.</p>
     </emu-note>
     <emu-alg>
+      1. If _templateOrFn_ is not an object, then
+        1. NOTE: This check is to allow future extensions with different input types.
+        1. Throw a *TypeError* exception.
       1. If IsCallable(_templateOrFn_) is *true*, then
         1. Let _tag_ be _templateOrFn_.
         1. Let _closure_ be a new Abstract Closure with parameters (_template_, ..._substitutions_) that captures _tag_ and performs the following steps when called:
+          1. If _template_ is not an object, then
+            1. Throw a *TypeError* exception.
           1. Let _R_ be the *this* value.
           1. Let _dedented_ be ? DedentTemplateStringsArray(_template_).
```

Comment:
`DedentTemplateStringsArray`'s `template` parameter is now always an Object, so the `ToObject` call in that AO can be dropped. Alternatively, we may throw the `TypeError` from `DedentTemplateStringsArray`, which makes the type validation a bit more implicit, but avoids duplication.

Reply:
Done.